File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/AccessProperties.swift
@main
struct MyMain {

    static func main() {
        let vm = VideoMode()
        let res = Resolution()
        let fr = vm.frameRate
        let px = res.height * res.width

    }
}
class VideoMode {
    var resolution = Resolution()
    var interlaced = false
    var frameRate = 0.0
    var name: String?
}

struct Resolution {
    var width = 0
    var height = 0
}

File: AutoTranspiledTests/Swift-to-Cpp/try-1/AccessProperties.cpp

#include <string>

struct Resolution {
    int width = 0;
    int height = 0;
};

struct VideoMode {
    Resolution resolution = Resolution();
    bool interlaced = false;
    double frameRate = 0.0;
    std::string name = "";
};

int main(){
	VideoMode vm;
	Resolution res;
	double fr = vm.frameRate;
	int px = res.height * res.width;

	return 0;
}